src/packets.toit

```diff
@@ -229,17 +229,20 @@ class SubscribePacket extends Packet:
 class SubAckPacket extends Packet implements PacketIDAck:
   static TYPE ::= 9
 
-  packet_id/int
-  qos/int
+  packet_id /int
+  qos /List  // The list of qos matches the list of topics from the SubPacket.
 
   constructor .packet_id --.qos:
     super TYPE
 
-  constructor.deserialize reader/reader.BufferedReader:
+  constructor.deserialize reader/reader.BufferedReader size/int:
     data := reader.read_bytes 2
     packet_id = binary.BIG_ENDIAN.uint16 data 0
-    qos = reader.read_byte
-    if qos == 127: throw "QoS negotiation failed"
+    consumed := 0
+    qos = []
+    while consumed < size:
```

Comment:
Where is `consumed` supposed to be incremented?

Reply:
done.